.github/workflows/build-test.yml

```diff
@@ -0,0 +1,180 @@
+name: test
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+        - '**'
+  pull_request:
+    branches:
+      - '**'
+
+concurrency:
+  # SHA is added to the end if on `main` to let all main workflows run
+  group: ${{ github.ref }}-${{ github.workflow }}-${{ github.event_name }}-${{ github.ref == 'refs/heads/main' && github.sha || '' }}
+  cancel-in-progress: true
+
+jobs:
+  configure:
+    name: Configure
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Setup Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Find test paths
+        id: configure
+        run: |
+          python tests/build-job-matrix.py --per directory --verbose > matrix.json
+          cat matrix.json
+          echo ::set-output name=configuration::$(cat matrix.json)
+    outputs:
+      configuration: ${{ steps.configure.outputs.configuration }}
+
+  test:
+    # Should match JOB_NAME below
+    name: ${{ matrix.configuration.name }} ${{ matrix.os.emoji }} ${{ matrix.python.name }}
+    needs: configure
+    runs-on: ${{ matrix.os.runs-on }}
+    timeout-minutes: ${{ matrix.configuration.job_timeout }}
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: ${{ fromJson(needs.configure.outputs.configuration) }}
+        os:
+          - emoji: 🍎
```

Comment:
This is smart.

Reply:
Woah, that's :banana:s

Reply:
So a downside here is that it'll be annoying to search for the workflow you want.  Which I have been doing a lot.  `:|`